### PR TITLE
Modifications for SMASH-vHLLE-hybrid sampling

### PR DESCRIPTION
**Pull request by NGoetz**
This PR:

- Adds the option to have bulk corrections for sampling
- Modifies the SMASH particle list in order to remove the sigma 9000221 (following https://inspirehep.net/files/e69bc5973b59fe900b46a702e9fa1702 Appendix G1)
- Fixes the output to be readable by the SMASH Afterburner setup

### CODE REVIEW

**Comment by NGoetz** [comment]
@vlvovch I cannot request you as a reviewer - could you take a look at this? Also the related PR in the sampler.

**Comment by vlvovch** [comment]
Thank you, I will check this out in the coming days.

File: src/library/HRGEventGenerator/HypersurfaceSampler.cpp [hunk 1]
```diff
@@ -635,16 +641,24 @@ namespace thermalfist {
         }
         // this is in principle the ansatz which is also used in https://github.com/smash-transport/smash-hadron-sampler
         // from this paper Phys.Rev.C 73 (2006) 064903
-        Weight_visc = (1.0 + (1.0 + stat * feq) * pipp / (2.0 * T * T * (elem->edens + elem->P)));
+        Weight_visc += ((1.0 + stat * feq) * pipp / (2.0 * T * T * (elem->edens + elem->P)));
+        
+      }
+      if (bulk_correction){
+        // this is in principle the ansatz which is also used in https://github.com/smash-transport/smash-hadron-sampler
+        // eq (7) of https://arxiv.org/pdf/1509.06738
```
Comment (vlvovch) [inline src/library/HRGEventGenerator/HypersurfaceSampler.cpp:649]: Plus Eq. (4) from https://arxiv.org/pdf/1403.0962 for \tau_Pi

File: src/library/HRGEventGenerator/HypersurfaceSampler.cpp [hunk 2]
```diff
@@ -635,16 +641,24 @@ namespace thermalfist {
...
+         double mom[4] = {part.p0, part.px, part.py, part.pz};
+         Weight_visc -= (1.0+stat*feq)*elem->Pi*(mass*mass/(3*mom[0])-mom[0]*(1.0/3.0-speed_of_sound_squared))//what to do about this
```
Comment (vlvovch) [inline src/library/HRGEventGenerator/HypersurfaceSampler.cpp:651]: This is the expression from Iurii's [note](https://github.com/smash-transport/smash-hadron-sampler/files/14011233/bulk_corrections_note.pdf)

File: include/HRGEventGenerator/HypersurfaceSampler.h [hunk 3]
```diff
@@ -202,12 +211,14 @@ namespace thermalfist {
     HypersurfaceEventGenerator(
       const ParticlizationHypersurface* hypersurface = NULL,
       ThermalModelBase* model = NULL,
-      double etasmear = 0.0, bool shear_correction = false) : EventGeneratorBase()
+      double etasmear = 0.0, bool shear_correction = false, bool bulk_correction = false, double speed_of_sound_squared = 0.333) : EventGeneratorBase()
```
Comment (vlvovch) [inline include/HRGEventGenerator/HypersurfaceSampler.h:214]: Not sure if 0.333 is the best default value, given the possible division by zero, namely (1/3 - cs^2), when computing \delta f_bulk

**Comment by vlvovch** [comment]
@NGoetz Thanks for the modifications. I've made some adjustments:

- Wrapped shear/bulk viscous correction flags, as well as etaSmear and speedOfSoundSquared parameters into a struct for better readability and flexibility for future extensions
- Changed default value of c_s^2 to from 0.33 to 0.15 to avoid division by (almost) zero
- Minor tidying up of the code

Please check that the code works as intended with these changes.

Another thing: bulk viscous corrections modify not only the shape of the momentum spectrum but also the multiplicities. Currently, this is not implemented in fist-sampler, or in smash-hadron-sampler in general(?). We may want to note this for the future.

**Comment by NGoetz** [comment]
I agree to the changes. Indeed, so far both only include modifications for the momentum spectrum. In future it would be good to also include the multiplicities.

**Comment by vlvovch** [comment]
Looks good. Merging.

**Pull request merged**